---
title: "Searching and Comparing PubMed Article Using R"
subtitle: "A simple practical user guides for publication statistics"
author: "<strong>Teresia Mrema Buza</strong>"
date: "Updated on `r Sys.Date()`"
site: bookdown::bookdown_site
knit: bookdown::render_book
output:
  bookdown::bs4_book:
    includes:
      in_header:
    template: bs4_template.html
documentclass: book
classoption: openany #remove empty pages in pdf doc
colorlinks: true
css:
- style.css
- css/style.css
always_allow_html: true
fig_caption: true
fontsize: 12pt
geometry: margin=1in
indent: false
keep_tex: true
link-citations: true
mainfont: Times New Roman
biblio-style: apalike
description: |
  Tracking and comparing articles published in the NCBI PubMed Database.
---

# PubMed Publications Tracking {#pubmed-search}

```{r include=FALSE}
library(DiagrammeR)
library(DiagrammeRsvg)
library(rentrez)
library(glue)
library(tidyverse, suppressPackageStartupMessages)
library(tictoc)
library(xlsx)

knitr::opts_chunk$set(
  echo = TRUE,
  message = FALSE,
  warning = FALSE,
  cache = FALSE,
  comment = NA,
  fig.path='./figures/',
  fig.show='asis',
  dev = 'png',
  fig.align='center',
  out.width = "70%",
  fig.width = 7,
  fig.asp = 0.7,
  fig.show = "asis"
)
```

## General Workflow
```{r echo=FALSE}
mermaidgraph <- mermaid("graph TD

subgraph  
A[<b>SEARCHING PUBMED ARTICLES</b>] -->  |Searching<br>Manually| B[Create Search Terms]
B --> C[NCBI PubMed Database] 
C --> |Results| C1[Indexed Papers in PubMed]
C1 --> |Download results| D[Results by Year in CSV format]
D --> |Filename| E[PubMed_Timeline_Results_by_Year.csv]
E --> F[Save the file in the data directory]
F -.-> G[Further Analysis in R<br>Such as plotting the results]


A --> |Seaching <br>Using a Code| H[Import Basic Libraries<li>rentrez</li><li>glue</li><li>tidyverse</li>]
H --> I[Define Search Terms]
I --> |Search counts| J[Computationally Get Search Counts]
J --> |Result<br> Manipulation| K[Tidying, Filtering, Transformation]
K --> L[Clean Data]
L --> F

end

", height = 800, width = 500)

mermaidgraph

```

## Structure of this guide {-}
This guide is divided into sections to facilitate easy navigation. The main chapter contains several sections as displayed in the navigation bars on the left and right. Click the hyper-linked text if you want to jump into a specific chapter or section.

## Code availability {-}
The code that generated this guide is available at a public [GitHub repository](https://github.com/tmbuza/pubmed-article-stats) or via a [GH-Page](https://tmbuza.github.io/pubmed-article-stats/). If interested you can request a consulting service by contacting the developer of this repo using <a href="https://complexdatainsights.com/contact-us">this contact form</a>.






<!--chapter:end:index.Rmd-->

# Defining Seach Terms
```{r}
year <- 1982:2022
microbiome_search <- glue("microbiome OR metagenomics OR metagenome AND {year}[PDAT]")
genomics_search <- glue("genomics AND {year}[PDAT]")
proteomics_search <- glue("proteomics AND {year}[PDAT]")
metabolomics_search <- glue("metabolomics AND {year}[PDAT]")
lipidomics_search <- glue("lipidomics AND {year}[PDAT]")
phenomics_search <- glue("phenomics AND {year}[PDAT]")
transcriptomics_search <- glue("transcriptomics AND {year}[PDAT]")
ml_search <- glue("(machine learning) AND {year}[PDAT]")
bioinfo_search <- glue("(bioinformatic*) AND {year}[PDAT]")
pharma_search <- glue("(pharmacogenomics*) AND {year}[PDAT]")
all_search <- glue("{year}[PDAT]")
```

# Getting Article Counts
```{r}
library(tictoc)
tic()

search_counts <- tibble(year = year,
       microbiome_search = microbiome_search,
       genomics_search = genomics_search,
       proteomics_search = proteomics_search,
       metabolomics_search = metabolomics_search,
       lipidomics_search = lipidomics_search,
       phenomics_search = phenomics_search,
       transcriptomics_search = transcriptomics_search,
       ml_search = ml_search,
       bioinfo_search = bioinfo_search,
       pharma_search = pharma_search,
       all_search = all_search) %>%
  mutate(
         microb_res = map_dbl(microbiome_search, ~entrez_search(db="pubmed", term=.x)$count),
         genom_res = map_dbl(genomics_search, ~entrez_search(db="pubmed", term=.x)$count),
         proteo_res = map_dbl(proteomics_search, ~entrez_search(db="pubmed", term=.x)$count),
         metabol_res = map_dbl(metabolomics_search, ~entrez_search(db="pubmed", term=.x)$count),
         lipid_res = map_dbl(lipidomics_search, ~entrez_search(db="pubmed", term=.x)$count),
         pheno_res = map_dbl(phenomics_search, ~entrez_search(db="pubmed", term=.x)$count),
         transcr_res = map_dbl(transcriptomics_search, ~entrez_search(db="pubmed", term=.x)$count),
         ml_res = map_dbl(ml_search, ~entrez_search(db="pubmed", term=.x)$count),
         bioinfo_res = map_dbl(bioinfo_search, ~entrez_search(db="pubmed", term=.x)$count),
         pharma_res = map_dbl(pharma_search, ~entrez_search(db="pubmed", term=.x)$count),
         all_res = map_dbl(all_search, ~entrez_search(db="pubmed", term=.x)$count)
         ) 

paste("The search took", toc())
```

## Saving results in multiple formats
- CSV
- XLSX
- RDS

```{r}
write_csv(search_counts, "data/search_counts.csv", col_names = TRUE)
write.xlsx(search_counts, "data/search_counts.xlsx", col_names = TRUE)
saveRDS(search_counts, "data/search_counts.rds", col_names = TRUE)
```

<!--chapter:end:01_search_pubmed.Rmd-->

# Tidying, Filtering, Transformation, and Plotting
```{r search_plot}
read_csv("data/search_counts.csv", show_col_types = TRUE) %>% 
  select(year, ends_with("_res")) %>%  
  filter(year >= 1991) %>% # Filter as you please!
  mutate(
    Microbiome = 100 * (microb_res / all_res),
    Genomics = 100 * (genom_res / all_res),
    Proteomics = 100 * (proteo_res / all_res),
    Metabolomics = 100 * (metabol_res / all_res),
    Lipidomics = 100 * (lipid_res / all_res),
    Phenomics = 100 * (pheno_res / all_res),
    Transcriptomics = 100 * (transcr_res / all_res),
    MachineLearning = 100 * (ml_res / all_res),
    Bioinformatics = 100 * (bioinfo_res / all_res),
    Pharmacogenomics = 100 * (pharma_res / all_res),
    AllArticles = all_res) %>% 
  select(-AllArticles, -ends_with("_res")) %>% 
  pivot_longer(-year) %>%
  filter(value > 0) %>% 
  ggplot(aes(x = year, y = value, group = name, color = name)) +
  geom_line(size = 1.2) +
  scale_color_manual(name = "FIELD",
                     values = c("red", "green4", "gray", "orange", "blue4", "green2", "magenta", "purple", "maroon", "blue1")) +
  labs(x = "Year", y = "Percentage of articles in PubMed", color = "FIELD") +
  theme_classic()

ggsave("figures/pubmed_search_line_plot.png", width = 8, height = 5)
# ggsave("../TMB-portfolio/figures/pubmed_search_line_plot.png", width = 8, height = 5)

```

<!--chapter:end:02_plot_article_count.Rmd-->

`r if (knitr::is_html_output()) '
# References {-}
'`

<!--chapter:end:999-references.Rmd-->

